src/lib.rs

```diff
@@ -29,6 +29,18 @@ use ark_std::{
     vec::Vec,
 };
 
+#[cfg(doctest)]
+mod test_readme {
+    macro_rules! external_doc_test {
+        ($x:expr) => {
+            #[doc = $x]
+            extern "C" {}
+        };
+    }
+
+    external_doc_test!(include_str!("../README.md"));
+}
```

Comment:
We can just include this in the crate documentation, like we do in algebra, no?

Reply:
Hm I thought `cargo doc test` would test it? (That was the entire point of including it in `algebra`, I believe)

Reply:
Indeed it does (but you have to invoke `cargo test --doc`)